homeassistant/components/teslemetry/__init__.py

```diff
@@ -42,29 +43,48 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         raise ConfigEntryNotReady from e
 
     # Create array of classes
-    data = []
+    vehicles: list[TeslemetryVehicleData] = []
+    energysites: list[TeslemetryEnergyData] = []
     for product in products:
-        if "vin" not in product:
-            continue
-        vin = product["vin"]
-
-        api = VehicleSpecific(teslemetry.vehicle, vin)
-        coordinator = TeslemetryVehicleDataCoordinator(hass, api)
-        data.append(
-            TeslemetryVehicleData(
-                api=api,
-                coordinator=coordinator,
-                vin=vin,
+        if "vin" in product:
+            vin = product["vin"]
+            api = VehicleSpecific(teslemetry.vehicle, vin)
+            coordinator = TeslemetryVehicleDataCoordinator(hass, api)
+            vehicles.append(
+                TeslemetryVehicleData(
+                    api=api,
+                    coordinator=coordinator,
+                    vin=vin,
+                )
+            )
+        elif "energy_site_id" in product:
```

Comment:
Just to be sure but the `product` can not contain both `vin` and `energy_site_id`?

Reply:
That is correct. It can't be a vehicle and an energy site.